File: app/src/main/java/com/ndmsystems/coala/layers/security/session/SecuredSessionPool.java
package com.ndmsystems.coala.layers.security.session;


import com.googlecode.concurrentlinkedhashmap.ConcurrentLinkedHashMap;

public class SecuredSessionPool {
    private ConcurrentLinkedHashMap<String, SecuredSession> pool;

    public SecuredSessionPool() {
        pool = new ConcurrentLinkedHashMap.Builder<String, SecuredSession>().maximumWeightedCapacity(500).build();
    }


    public void set(String hash, SecuredSession session) {
        if (hash != null) {
            pool.put(hash, session);
        }
    }

    public SecuredSession get(String hash) {
        if (hash == null) {
            return null;
        }

        return pool.get(hash);
    }

    public SecuredSession getByPeerProxySecurityId(Long peerProxySecurityId) {
        if (peerProxySecurityId == null) {
            return null;
        }

        for (SecuredSession session : pool.values()) {
            if(session != null && peerProxySecurityId.equals(session.getPeerProxySecurityId())){
                return session;
            }
        }

        return null;
    }

    public void remove(String hash) {
        pool.remove(hash);
    }

    public void clear() {
        pool.clear();
    }
}

File: app/src/main/java/com/ndmsystems/coala/layers/security/session/SecuredSessionPool.kt
package com.ndmsystems.coala.layers.security.session

import com.googlecode.concurrentlinkedhashmap.ConcurrentLinkedHashMap

class SecuredSessionPool {
    private val pool: ConcurrentLinkedHashMap<String, SecuredSession> =
        ConcurrentLinkedHashMap.Builder<String, SecuredSession>().maximumWeightedCapacity(500).build()

    operator fun set(hash: String?, session: SecuredSession) {
        if (hash != null) {
            pool[hash] = session
        }
    }

    operator fun get(hash: String?): SecuredSession? {
        return if (hash == null) {
            null
        } else pool[hash]
    }

    fun getByPeerProxySecurityId(peerProxySecurityId: Long?): SecuredSession? {
        if (peerProxySecurityId == null) {
            return null
        }
        for (session in pool.values) {
            if (session != null && peerProxySecurityId == session.peerProxySecurityId) {
                return session
            }
        }
        return null
    }

    fun remove(hash: String) {
        pool.remove(hash)
    }

    fun clear() {
        pool.clear()
    }
}